src/posthog-chat.ts

```diff
@@ -0,0 +1,133 @@
+                    title: 'Chat widget',
+                    message: message,
+                    source_url: 'chat-widget',
+                },
+                timeout: 10000,
+                callback: (response) => {
+                    const statusCode = response.statusCode
+                    if (statusCode !== 200 || !response.json) {
+                        const error = `Chat message could not be sent, status: ${statusCode}`
+                        logger.error(error)
+                    }
+
+                    this.getChat()
+                },
+            })
+        } else {
+            this._instance._send_request({
+                url: this._instance.requestRouter.endpointFor('api', `/api/chat/`),
+                method: 'POST',
+                data: {
+                    token: this._instance.config.token,
+                    action: 'send_message',
+                    conversation_id: conversationId,
+                    message: message,
+                    distinct_id: this._instance.get_distinct_id(),
+                },
+                timeout: 10000,
+                callback: (response) => {
+                    const statusCode = response.statusCode
+                    if (statusCode !== 200 || !response.json) {
+                        const error = `Chat message could not be sent, status: ${statusCode}`
+                        logger.error(error)
+                    }
+
+                    this.getChat()
+                },
+            })
+        }
+    }
+
+    getChat() {
+        /** No calls if chat is not enabled */
+        if (!this.isEnabled) {
+            return
+        }
+
+        try {
+            this._instance._send_request({
+                url: this._instance.requestRouter.endpointFor(
+                    'api',
+                    `/api/chat/?token=${this._instance.config.token}&distinct_id=${this._instance.get_distinct_id()}`
+                ),
+                method: 'GET',
+                timeout: 10000,
+                callback: (response) => {
+                    const statusCode = response.statusCode
+                    if (statusCode !== 200 || !response.json) {
+                        const error = `Chat API could not be loaded, status: ${statusCode}`
+                        logger.error(error)
+                        // return callback([], {
+                        //     isLoaded: false,
+                        //     error,
+                        // })
+                    }
+                    const chats = response.json.conversations || []
+
+                    if (chats.length > 0) {
+                        const chat = chats[0]
+                        this.messages = chat.messages || []
+                        this.conversationId = chat.id
+                    }
+                },
+            })
+        } catch (e) {
+            logger.error('PostHogChat getChat', e)
+            throw e
```

Comment:
i wouldn't throw as well, risk that the SDK breaks the customer site or itself

we do have an onError callback i think, but i think it's also focussed on capture 🤔 
(something that generally our SDKS could be better at tbh)